pkg/kafka/auth/scram.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"crypto/sha512"
+	"hash"
+
+	"github.com/Shopify/sarama"
+	"github.com/pkg/errors"
+	scrampkg "github.com/xdg/scram"
+)
+
+// SHA512 hash setting
+var SHA512 scrampkg.HashGeneratorFcn = func() hash.Hash { return sha512.New() }
+
+// ScramConfig describes the configuration properties required for the SCRAM handshake
+// between the collector and the pipeline
+type ScramConfig struct {
+	UserName  string `mapstructure:"username"`
+	Password  string `mapstructure:"password"`
+	Algorithm string `mapstructure:"algorithm"`
+}
+
+// SetSCRAMConfiguration ...
+func setSCRAMConfiguration(config *ScramConfig, saramaConfig *sarama.Config) error {
+	fn := func() sarama.SCRAMClient {
+		return &scramClient{}
+	}
+
+	var mechanism sarama.SASLMechanism
+
+	switch config.Algorithm {
+	case "sha512":
+		fn = func() sarama.SCRAMClient {
+			return &scramClient{
+				HashGeneratorFcn: SHA512,
+			}
+		}
+		mechanism = sarama.SASLMechanism(sarama.SASLTypeSCRAMSHA512)
+	case "sha256":
+		fn = func() sarama.SCRAMClient {
+			return &scramClient{
+				HashGeneratorFcn: scrampkg.SHA256,
+			}
+		}
+		mechanism = sarama.SASLMechanism(sarama.SASLTypeSCRAMSHA256)
+	default:
+		return errors.Errorf("invalid SHA algorithm '%s': can be either 'sha256' or 'sha512'", config.Algorithm)
```

Comment:
we don't use pkg/errors, this can be simply `fmt.Errorf`

Reply:
Changed!